### ISSUE TITLE
Fix FP on S3402 with computed ambient enums (TypeScript)

### ISSUE THREAD

**Issue by guillaume-dequenne-sonarsource** [SonarSource/SonarJS#2171]
Reported by community: https://community.sonarsource.com/t/ambient-computed-enum-value-is-incorrectly-not-typed-as-a-number/30627

The following snippet raises an FP on S3402:
```
declare enum E { X, Y }
const a = 1 + E.X;
```

In this case, the value `E.X`s computed to be a number and therefore should not trigger S3402.

**Comment by andrea-guarino-sonarsource** [comment]
The issue is no reproducible after migration implementation to `typescript-eslint`
